### PR TITLE
Fix topLevelDomain() for IDN hosts

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix topLevelDomain() for IDN hosts (i.e. `example.рф`), before it returns empty string for such hosts.

### CODE REVIEW

File: src/Functions/URL/topLevelDomain.cpp [hunk 1]
```diff
@@ -28,7 +28,7 @@ struct ExtractTopLevelDomain
                 return;
 
             /// For IPv4 addresses select nothing.
-            if (last_dot[1] <= '9')
+            if (last_dot[1] >= '0' && last_dot[1] <= '9')
```
Comment (alexey-milovidov) [inline src/Functions/URL/topLevelDomain.cpp:31]: isNumericASCII

Comment (alexey-milovidov) [reply src/Functions/URL/topLevelDomain.cpp:31]: Maybe also add
- a comment that it expects null terminated string?

**Comment by alexey-milovidov** [comment]
One test reference has to be updated.

**Comment by alexey-milovidov** [comment]
The changelog entry must be readable for ClickHouse users.

File: src/Functions/URL/topLevelDomain.cpp (hunk 1 above)
Comment (azat) [reply src/Functions/URL/topLevelDomain.cpp]: > a comment that it expects null terminated string?

Added another comment (about getURLHost())

>isNumericASCII

Right, done

**Comment by alexey-milovidov** [comment]
@tavplubix Need to check 00993_system_parts_race_condition_drop_zookeeper

**Comment by alexey-milovidov** [comment]
Performance test is Ok.

**Comment by tavplubix** [comment]
> @tavplubix Need to check 00993_system_parts_race_condition_drop_zookeeper

Seems like `StorageReplicatedMergeTree::shutdown()` hung somewhere between `part_moves_between_shards_orchestrator.shutdown()` and `clearOldPartsFromFilesystem(true);`:
```
$ zgrep -Fa "2d08b532-f80b-47c4-995e-164075c57caa" clickhouse-server1.log.gz25103 
2021.06.09 13:14:06.543195 [ 873 ] {2d08b532-f80b-47c4-995e-164075c57caa} <Debug> executeQuery: (from 0.0.0.0:0, user: , using production parser) (comment: '/usr/share/clickhouse-test/queries/0_stateless/00993_system_parts_race_condition_drop_zookeeper.sh') /* ddl_entry=query-0000000060 */ DROP TABLE IF EXISTS test_z3tqi0.alter_table_1 NO DELAY
2021.06.09 13:14:06.543250 [ 873 ] {2d08b532-f80b-47c4-995e-164075c57caa} <Trace> test_z3tqi0.alter_table_1 (ReplicatedMergeTreeRestartingThread): Restarting thread finished
2021.06.09 13:14:06.544325 [ 873 ] {2d08b532-f80b-47c4-995e-164075c57caa} <Trace> test_z3tqi0.alter_table_1 (ReplicatedMergeTreeRestartingThread): Waiting for threads to finish
2021.06.09 13:14:06.544339 [ 873 ] {2d08b532-f80b-47c4-995e-164075c57caa} <Information> test_z3tqi0.alter_table_1 (030c98e1-2dba-47c2-830c-98e12dba77c2): Stopped being leader
2021.06.09 13:14:06.561794 [ 873 ] {2d08b532-f80b-47c4-995e-164075c57caa} <Trace> test_z3tqi0.alter_table_1 (ReplicatedMergeTreeRestartingThread): Threads finished
2021.06.09 13:14:06.561820 [ 873 ] {2d08b532-f80b-47c4-995e-164075c57caa} <Trace> test_z3tqi0.alter_table_1 (PartMovesBetweenShardsOrchestrator): PartMovesBetweenShardsOrchestrator thread finished
```

**Pull request merged**